weldx/util.py

```diff
@@ -1201,3 +1193,78 @@ def time_ref(self, value: pd.Timestamp):
                 self._obj.time.attrs["time_ref"] = value  # set new time_ref value
             else:
                 self._obj.time.attrs["time_ref"] = value
+
+
+class _Eq_compare_nested:
+    """Compares nested data structures like lists, sets, tuples, arrays, etc."""
+
+    compare_funcs = {
+        (np.ndarray, pint.Quantity): lambda x, y: np.all(x == y),
+        (xr.DataArray, xr.Dataset): lambda x, y: x.identical(y),
+    }
+
+    @staticmethod
+    def _compare(x, y):
+        if type(x) is not type(y):
+            return False
```

Comment:
```suggestion
        x, y = (np.asarray(v) if isinstance(v, NDArrayType) else v for v in (x, y))
        if type(x) is not type(y):
            return False
```
When reading ASDF trees numpy arrays get loaded as `NDArrayType` which can be confusing
This should be a simple fix to catch that case

Reply:
The asdf tree test cases did not fail for NDArrayType, so I guess this is fine.

Reply:
I think this is because the example does not include "raw" numpy arrays that ar enot nested in dataclasses/xarray or simliar

Reply:
Does ASDF perform this wrapping only, if arrays are being red lazily?

Reply:
good question, I did not look into it that deeply but it sounds reasonable